doc/dev/ARCHITECTURE.md

```diff
@@ -0,0 +1,68 @@
+# F3D Architecture
+
+F3D is structured in different parts, interacting with each others and with F3D dependencies.
+This architecture is reflected by the directories organisation.
+
+- **application**: the code of the F3D application itself, see below
+- cmake: cmake macros and functions, used by the CMake build system
+- doc: this very documentation
+- examples: examples usage of the libf3d and plugin framework in python and C++
+- external: dependencies that are included directly in the code
+- java: [java bindings](../libf3d/LANGUAGE_BINDINGS.md#java-experimental) and associated tests.
+- **library**: the [libf3d](../libf3d/README_LIBF3D.md) itself, see below
+- **plugins**: all the [plugins](../libf3d/PLUGINS.md) providing different readers, see below
+- python: [python bindings](../libf3d/LANGUAGE_BINDINGS.md#python) and tests
+- resources: all non code, non doc, like icon, configs and such
+- testing: all testing related resources, does not contain the test themselves
+- **vtkext**: extensions to VTK and related tests, see below
+- webassembly: [webassembly/javascript bindings](../libf3d/LANGUAGE_BINDINGS.md#javascript-experimental) and [F3DWeb](https://f3d.app/web/) application code
+- winshellext: shell extension for Windows, provide [thumbnails for Windows](../user/DESKTOP_INTEGRATION.md#windows)
+
+Here is diagram explaining how some of these parts interact together:
+
+<img src="https://media.githubusercontent.com/media/f3d-app/f3d-media/c43c1367a2aac4c9a3bdd478bdac3ffefe82750e/media/archi.png" width="700">
+
+## vtkext
+
+`vtkext` contains two [VTK modules](https://docs.vtk.org/en/latest/api/cmake/ModuleSystem.html) that are used extensively in the libf3d and the plugins.
+
+`public` is a VTK module that contains classes and utilities that can be installed and used by plugins, including externals plugins. `vtkF3DImporter` is a class
```

Comment:
Please specify the public is part of the SDK, but the private one is not.

Reply:
add a mention of `plugin_sdk` but im not sure this is what you had in mind